crates/nu-command/src/filesystem/cp.rs

```diff
@@ -372,239 +94,141 @@ impl Command for Cp {
+                Some(target.span),
+                None,
+                Vec::new(),
+            ));
+        };
+        // paths now contains the sources
+        let sources: Vec<Vec<PathBuf>> = paths
+            .iter()
+            .map(|p| {
+                // Need to expand too make it work with globbing
+                let expanded_src = expand_to_real_path(&p.item);
+                match nu_glob::glob_with(&expanded_src.to_string_lossy(), GLOB_PARAMS) {
+                    Ok(files) => {
+                        let f = files.filter_map(Result::ok).collect::<Vec<PathBuf>>();
+                        if f.is_empty() {
+                            return Err(ShellError::FileNotFound(p.span));
                         }
+                        Ok(f)
                     }
-                    Err(e) => {
-                        // There was a problem writing the dst file
-                        process_failed = Some(e);
-                        break;
-                    }
+                    Err(e) => Err(ShellError::GenericError(
+                        e.to_string(),
+                        "invalid pattern".to_string(),
+                        Some(p.span),
+                        None,
+                        Vec::new(),
+                    )),
                 }
-            }
-            Err(e) => {
-                // There was a problem reading the src file
-                process_failed = Some(e);
-                break;
-            }
+            })
+            .collect::<Result<Vec<Vec<PathBuf>>, ShellError>>()?;
+
+        let sources = sources.into_iter().flatten().collect::<Vec<PathBuf>>();
+        let options = uu_cp::Options {
+            attributes_only: false,
+            backup: BackupMode::NoBackup,
+            copy_contents: false,
+            cli_dereference: false,
+            copy_mode: uu_cp::CopyMode::Copy,
+            // dereference,
+            dereference: !recursive,
+            no_target_dir: false,
+            one_file_system: false,
+            overwrite,
+            parents: false,
+            sparse_mode: uu_cp::SparseMode::Auto,
+            strip_trailing_slashes: false,
+            reflink_mode,
+            attributes: uu_cp::Attributes::NONE,
+            recursive,
+            backup_suffix: String::from("~"), // default
+            target_dir: None,
+            update: UpdateMode::ReplaceAll,
+            debug,
+            verbose: verbose || debug,
+            progress_bar: progress,
         };
-    }
 
-    // If copying the file failed
-    if let Some(error) = process_failed {
-        if error.kind() == ErrorKind::Interrupted {
-            bar.abandoned_msg("# !! Interrupted !!".to_owned());
-        } else {
-            bar.abandoned_msg("# !! Error !!".to_owned());
+        if let Err(error) = uu_cp::copy(&sources, &target_path, &options) {
+            match error {
+                // code should still be EXIT_ERR as does GNU cp
```

Comment:
We could think about this actually, maybe that's not necessary, but we'll have to see in what exact cases this branch is triggered.

Reply:
Yeah, I think that part about the `set_exit_code` we have to figure it out perhaps over on a new PR, but i have left those comments as its a `TODO` for now